presentation-compiler/src/main/dotty/tools/pc/DiagnosticProvider.scala

```diff
@@ -0,0 +1,65 @@
+package dotty.tools.pc
+
+import org.eclipse.lsp4j
+import org.eclipse.lsp4j.DiagnosticSeverity
+import dotty.tools.dotc.interactive.InteractiveDriver
+import dotty.tools.dotc.interfaces.Diagnostic as DiagnosticInterfaces
+import dotty.tools.dotc.reporting.Diagnostic
+import dotty.tools.pc.utils.InteractiveEnrichments.toLsp
+
+import scala.meta.pc.VirtualFileParams
+import ch.epfl.scala.bsp4j
+import dotty.tools.dotc.reporting.CodeAction
+import dotty.tools.dotc.rewrites.Rewrites.ActionPatch
+import scala.jdk.CollectionConverters.*
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.reporting.ErrorMessageID
+import org.eclipse.lsp4j.DiagnosticTag
+
+class DiagnosticProvider(driver: InteractiveDriver, params: VirtualFileParams):
+
+  def diagnostics(): List[lsp4j.Diagnostic] =
+    val diags = driver.run(params.uri().nn, params.text().nn)
+    given Context = driver.currentCtx
+    diags.flatMap(toLsp)
+
+  private def toLsp(diag: Diagnostic)(using Context): Option[lsp4j.Diagnostic] =
+    Option.when(diag.pos.exists):
+      val lspDiag = lsp4j.Diagnostic(
+        diag.pos.toLsp,
+        diag.msg.message,
+        toDiagnosticSeverity(diag.level),
+        "presentation compiler",
+      )
+      lspDiag.setCode(diag.msg.errorId.errorNumber)
+
+      val scalaDiagnostic = new bsp4j.ScalaDiagnostic()
```

Comment:
Shouldn't we instead create our own diagnotic in metals and translate them to code actions as well? We wouldn't need to add bsp4j to the PC compiler.